### DISCUSSION TITLE
Tigo Solar Panel Optimisers

### DISCUSSION THREAD

**Discussion by MJWMJW2** [Koenkk/zigbee2mqtt#18020] [Devices]
Tigo solar panel optimisers use zigbee to communicate. 

The ‘hub’ and access they provide is poor, even though hub runs locally, they will not allow you access to it or any deep panel data or real time, only via their cloud portal with around half hour updates. 

Is there any appetite to reverse engineer to see if it can be brought in to zigbee2mqtt? 

I would be happy to sponsor an experienced developer with hardware costs if required.

**Comment by mikee47** [comment]
I agree this is less than ideal. However, the logs are useful and the annual subscription reasonable so it could be a lot worse.
The main issues I have with the CCA/TAP setup is:

1. If the CCA or TAP fails then panels will not operate (safety shutdown).
2. Tigo are free to push firmware updates out without any user knowledge (consent is probably buried in the user agreement).
3. Critical infrastructure should *not* be routinely connected to the internet, and should certainly not require it to operate. The only justification for this is to allow manufacturers to keep control over your system and make money off you. Unfortunately this is just reality nowadays.

Back to your question. Tigo clearly have no interest in providing encryption keys to access the Zigbee traffic so that's a non-starter. However, the traffic between CCA and TAP(s) is regular RS485 serial traffic at 38400 baud and can be easily sniffed. If sense can be made of that then there's your realtime feed.

I have a suspicion, though, that without a cloud connection the CCA won't monitor panel data as there's nowhere to send it, so it'd need to be kept live. (Thus negating all three points above.)

**Comment by MJWMJW2** [comment]
To reply to your points directly.

1. I've had this feature turned off.  I could not connect CCA to a UPS supply, and still would like the batteries to charge in case of grid fail.

2.  This in fact has caused me issues, and why the CCA is now blocked from the internet directly.  The CCA / TAP / Optimiser combination caused severe intereferance with my home automation zigbee network, causing many devices to 'panic' and reset.  Sockets would turn off and lights turn on to their default state as optimisers started up.  Some devices would eventually lock up, this was especially bad as they were typically routers.   This started directly after having the Tigo system installed.  I could replicate this at will by power cycling the CCA, https://www.youtube.com/shorts/6RVVekv1e-8  I resolved this by moving Z2M channel.  Tigo would continually blame everyting but their own device, despite significant evidence to show it was them causing interferance but did agree to drop their transmit power.

However the issue returned some months later, this time seemed to be much worse, with Z2M repeatedly crashing and restarting multiple times per day also.  Again I had to move Z2M channel which is now 4 days stable.  I suspect they performed an update and the transmit power went to default, or they changed Zigbee channel.

3. Agreed.

4. There is also runnig a local webserver on the CCA, it was trivial to access, but has since received security updates, and it seems so far no amateur has managed to bypass whatever security is in place.  There's some suggestion the CCA may be needed to keep the best optimisation, however there are also conflciting statements that it is not.   There are some beginnig of interesting information here, but it seems to have died : https://www.photovoltaikforum.com/thread/149592-details-protokolle-zugang-auf-tigo-cca/?pageNo=1 (translate works fine to English is needed) 

So for I wouild like to keep it runing in case of optimisation benefit, but mostly  to check in on the system, I cannot do this with it blocked from the internet, nor I do not have the skill to sniff or understand RS485.

**Comment by mikee47** [comment]
Thank you for the additional information. I've found your thread in Tigo support and read the Swiss articles on the interference. That it emanates from the DC-DC converter in the optimiser is, in hindsight, not surprising as there will be a lot of energy with 10A + going through the module.

It's likely that Tigo will have quietly improved the filtering on later model optimisers as they're clearly aware of the issue. Still, in your situation some additional filtering on the DC feed from the panels may help - if you haven't already, whacking some split-core ferrite cores close to where they come down from the roof would be cheap enough to try?

Thank you for the note on being able to disable the rapid-shutdown feature via Tigo customer support. I'm going to have to chew on that one myself - panel-level shutdown is a pretty significant plus for these devices.

Maybe one day Tigo will offer local access on a subscription basis?

**Comment by MJWMJW2** [comment]
I am not sure how you arrive at zigbee issues caused by the DC to DC and to use split core ferrites. This is actually a different issue reported by some to cause other radio wave interference. 

My issue is specifically around zigbee interference, which can be replicated at will by powering only the TAP (and is also caused by each module).  As mentioned this is solved (2nd time!) by changing zigbee channels in my home network. 

it would be nice to use the features of the system and read optimisation data from each optimiser if it was possible without Tigo cloud, and to be able to control which zigbee channel they were using so that everything worked in harmony.

However if it is not not actually compliant with zigbee home automation standards then there is no point considering it further. And I will have to accept that the control system will remain offline.

**Reply by tbsmith1** [reply]
A little late to the party, but I am one that had Tigo change the behavior of the system so it doesn't shut down on power failure.  Instead I can use a manual pushbutton to intiate shutdown. 

May I ask what Zigbee channel you used to reduce the interefence?  I am currently on Channel 15 with wife on Channel 11.  Very few nearby wifi networks and they are not very close.

Have you ever tried the split core filters on the leads from the roof? 

BTW, I came across this thread when trying to figure out why I get so much interfernce on my Zigbee network during daylight hours and am 95% sure it is from the Tigo TS4 modules ( I powered down the TAP and CCA but it had no effect - modules are still functioning). It could be something other than the TS4s but I've yet to discover what.
